controllers/nodefeaturediscovery_controls.go

```diff
@@ -41,25 +43,40 @@ const (
 	defaultServicePort int = 12000
 )
 
+// String returns the status of the resource as being Ready,
+// NotReady, or Unknown Resource Status
 func (s ResourceStatus) String() string {
 	names := [...]string{
 		"Ready",
 		"NotReady"}
 
+	// Ideally, 's' should be either Ready (=0) or NotReady (=1), but
+	// we may run into a case where we get an unknown status, so return
+	// information stating that the resource status is unknown
 	if s < Ready || s > NotReady {
 		return "Unkown Resources Status"
 	}
 	return names[s]
 }
 
+// Namespace checks if the Namespace for NFD exists and attempts to
+// create it if it doesn't exist
```

Comment:
I'd state that the function *creates* instead of *attempts to create*. I used to use the former expression (i.e. *attempts to...* or *tries to...*) a lot myself earler but was advised on a few occasions to get rid of it 😄  And I now think it's a good advice. Thus

```suggestion
// Namespace checks if the Namespace for NFD exists and creates it
// if it doesn't exist
```